### PR TITLE
Add devcontainer config for developing Gitea

### PR DESCRIPTION
**Pull request by techknowlogick**
With the recent release of https://github.com/loft-sh/devpod making ephemeral dev environments (ala GitPod/Codespaces) much easier (aka, no pre-configured infra required). This allows an alternative to using GitPod for such development options.

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      "version": "20"
+    }
+  },
+  "customizations": {
+		"vscode": {
+			"settings": {},
+      // same extensions as GitPod
+			"extensions": [
+				"editorconfig.editorconfig",
+        "dbaeumer.vscode-eslint",
```
Comment (wxiaoguang) [inline .devcontainer/devcontainer.json]: Tab/Space are mixed in this file. Should it be unified?

Comment (techknowlogick) [reply .devcontainer/devcontainer.json]: yup, copy/pasting from multiple locations. fixed now.

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  // installs nodejs into container
+  "features": {
+      "ghcr.io/devcontainers/features/node:1": {
+          "version": "20"
+      }
+  },
+  "customizations": {
+      "vscode": {
+          "settings": {},
+          // same extensions as GitPod
```
Comment (yardenshoham) [inline .devcontainer/devcontainer.json]:
```suggestion
          // same extensions as Gitpod, should match /.gitpod.yml
```

**Comment by yardenshoham** [comment]
In Gitpod we also write an `app.ini` and run make watch. I wonder if that's possible here, could also go in a future PR

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (silverwind) [reply .devcontainer/devcontainer.json]: You could add the file to editorconfig lint to prevent such issues in the future.

**Comment by silverwind** [comment]
Where is the start command?

**Comment by silverwind** [comment]
Also, please re-format this file to 2-space.

**Comment by techknowlogick** [comment]
> Also, please re-format this file to 2-space indent.

updated formatting.

> Where is the start command?

none yet.

**Comment by yardenshoham** [comment]
I tried spinning up a GitHub Codespace but it failed, log:

```log
###########                                                               15.4%2023-05-18 14:56:47.682Z: 
###########                                                               15.5%2023-05-18 14:56:48.734Z: 
###########                                                               15.6%2023-05-18 14:57:27.621Z: 
###########                                                               15.7%curl: (18) transfer closed with 21098780 bytes remaining to read
#17 122.9 
#17 122.9 Binary download from https://nodejs.org/dist/v20.2.0/node-v20.2.0-linux-x64.tar.xz failed, trying source.
#17 122.9 grep: /usr/local/share/nvm/.cache/bin/node-v20.2.0-linux-x64/node-v20.2.0-linux-x64.tar.xz: No such file or directory
#17 122.9 Provided file to checksum does not exist.
#17 122.9 Binary download failed, trying source.
#17 122.9 Detected that you have 2 CPU core(s)
2023-05-18 14:57:27.771Z: #17 122.9 Number of CPU core(s) less than or equal to 2, running in single-threaded mode
2023-05-18 14:57:27.922Z: #17 123.1 Downloading https://nodejs.org/dist/v20.2.0/node-v20.2.0.tar.xz...
2023-05-18 14:57:28.224Z: #17 123.4 #=#=#                                                                         
curl: (22) The requested URL returned error: 500 
#17 123.4 
#17 123.4 Binary download from https://nodejs.org/dist/v20.2.0/node-v20.2.0.tar.xz failed, trying source.
#17 123.4 grep: /usr/local/share/nvm/.cache/src/node-v20.2.0/node-v20.2.0.tar.xz: No such file or directory
#17 123.4 Provided file to checksum does not exist.
#17 123.4 ERROR: Feature "Node.js (via nvm), yarn and pnpm" (ghcr.io/devcontainers/features/node) failed to install! Look at the documentation at https://github.com/devcontainers/features/tree/main/src/node for help troubleshooting this error.
2023-05-18 14:57:28.243Z: {"outcome":"error","message":"Command failed: docker buildx build --load --build-context dev_containers_feature_content_source=/tmp/devcontainercli-root/container-features/0.41.0-1684421683984 --build-arg _DEV_CONTAINERS_BASE_IMAGE=mcr.microsoft.com/devcontainers/go:1.20 --build-arg _DEV_CONTAINERS_IMAGE_USER=root --build-arg _DEV_CONTAINERS_FEATURE_CONTENT_SOURCE=dev_container_feature_content_temp --target dev_containers_target_stage -t vsc-gitea-e34ec3f68ee66efeb08de38b5a53c8c0e83bc2a9a5802233221d193ce35354ef-features -f /tmp/devcontainercli-root/container-features/0.41.0-1684421683984/Dockerfile.extended /var/lib/docker/codespacemount/.persistedshare/empty-folder","description":"An error occurred setting up the container."}
2023-05-18 14:57:28.249Z: #17 ERROR: executor failed running [/bin/sh -c cp -ar /tmp/build-features-src/node_1 /tmp/dev-container-features  && chmod -R 0755 /tmp/dev-container-features/node_1  && cd /tmp/dev-container-features/node_1  && chmod +x ./devcontainer-features-install.sh  && ./devcontainer-features-install.sh  && rm -rf /tmp/dev-container-features/node_1]: exit code: 1

#9 [internal] load metadata for mcr.microsoft.com/devcontainers/go:1.20
------
 > [dev_containers_target_stage 5/5] RUN --mount=type=bind,from=dev_containers_feature_content_source,source=node_1,target=/tmp/build-features-src/node_1     cp -ar /tmp/build-features-src/node_1 /tmp/dev-container-features  && chmod -R 0755 /tmp/dev-container-features/node_1  && cd /tmp/dev-container-features/node_1  && chmod +x ./devcontainer-features-install.sh  && ./devcontainer-features-install.sh  && rm -rf /tmp/dev-container-features/node_1:
#17 122.9 Binary download failed, trying source.
#17 122.9 Detected that you have 2 CPU core(s)
#17 122.9 Number of CPU core(s) less than or equal to 2, running in single-threaded mode
#17 123.1 Downloading https://nodejs.org/dist/v20.2.0/node-v20.2.0.tar.xz...
#17 123.4 #=#=#                                                                         
curl: (22) The requested URL returned error: 500 
#17 123.4 
#17 123.4 Binary download from https://nodejs.org/dist/v20.2.0/node-v20.2.0.tar.xz failed, trying source.
#17 123.4 grep: /usr/local/share/nvm/.cache/src/node-v20.2.0/node-v20.2.0.tar.xz: No such file or directory
#17 123.4 Provided file to checksum does not exist.
#17 123.4 ERROR: Feature "Node.js (via nvm), yarn and pnpm" (ghcr.io/devcontainers/features/node) failed to install! Look at the documentation at https://github.com/devcontainers/features/tree/main/src/node for help troubleshooting this error.
------
ERROR: failed to solve: executor failed running [/bin/sh -c cp -ar /tmp/build-features-src/node_1 /tmp/dev-container-features  && chmod -R 0755 /tmp/dev-container-features/node_1  && cd /tmp/dev-container-features/node_1  && chmod +x ./devcontainer-features-install.sh  && ./devcontainer-features-install.sh  && rm -rf /tmp/dev-container-features/node_1]: exit code: 1
2023-05-18 14:57:28.252Z: Stop: Run: docker buildx build --load --build-context dev_containers_feature_content_source=/tmp/devcontainercli-root/container-features/0.41.0-1684421683984 --build-arg _DEV_CONTAINERS_BASE_IMAGE=mcr.microsoft.com/devcontainers/go:1.20 --build-arg _DEV_CONTAINERS_IMAGE_USER=root --build-arg _DEV_CONTAINERS_FEATURE_CONTENT_SOURCE=dev_container_feature_content_temp --target dev_containers_target_stage -t vsc-gitea-e34ec3f68ee66efeb08de38b5a53c8c0e83bc2a9a5802233221d193ce35354ef-features -f /tmp/devcontainercli-root/container-features/0.41.0-1684421683984/Dockerfile.extended /var/lib/docker/codespacemount/.persistedshare/empty-folder
2023-05-18 14:57:28.346Z: Error: Command failed: docker buildx build --load --build-context dev_containers_feature_content_source=/tmp/devcontainercli-root/container-features/0.41.0-1684421683984 --build-arg _DEV_CONTAINERS_BASE_IMAGE=mcr.microsoft.com/devcontainers/go:1.20 --build-arg _DEV_CONTAINERS_IMAGE_USER=root --build-arg _DEV_CONTAINERS_FEATURE_CONTENT_SOURCE=dev_container_feature_content_temp --target dev_containers_target_stage -t vsc-gitea-e34ec3f68ee66efeb08de38b5a53c8c0e83bc2a9a5802233221d193ce35354ef-features -f /tmp/devcontainercli-root/container-features/0.41.0-1684421683984/Dockerfile.extended /var/lib/docker/codespacemount/.persistedshare/empty-folder
2023-05-18 14:57:28.399Z:     at Ose (/usr/lib/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:1946:3264)
2023-05-18 14:57:28.406Z:     at nO (/usr/lib/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:1946:3200)
2023-05-18 14:57:28.410Z:     at processTicksAndRejections (internal/process/task_queues.js:95:5)
2023-05-18 14:57:28.411Z:     at async Yse (/usr/lib/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:1961:2626)
2023-05-18 14:57:28.413Z:     at async vh (/usr/lib/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:1961:3741)
2023-05-18 14:57:28.416Z:     at async Eae (/usr/lib/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:2092:10371)
2023-05-18 14:57:28.419Z:     at async gae (/usr/lib/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:2092:10112)
2023-05-18 14:57:28.422Z: devcontainer process exited with exit code 1

====================================== ERROR ====================================
2023-05-18 14:57:28.427Z: Failed to create container.
=================================================================================
2023-05-18 14:57:28.429Z: Error: Command failed: docker buildx build --load --build-context dev_containers_feature_content_source=/tmp/devcontainercli-root/container-features/0.41.0-1684421683984 --build-arg _DEV_CONTAINERS_BASE_IMAGE=mcr.microsoft.com/devcontainers/go:1.20 --build-arg _DEV_CONTAINERS_IMAGE_USER=root --build-arg _DEV_CONTAINERS_FEATURE_CONTENT_SOURCE=dev_container_feature_content_temp --target dev_containers_target_stage -t vsc-gitea-e34ec3f68ee66efeb08de38b5a53c8c0e83bc2a9a5802233221d193ce35354ef-features -f /tmp/devcontainercli-root/container-features/0.41.0-1684421683984/Dockerfile.extended /var/lib/docker/codespacemount/.persistedshare/empty-folder
2023-05-18 14:57:28.431Z: Error Code: 1302

====================================== ERROR ====================================
2023-05-18 14:57:28.461Z: Container creation failed.
=================================================================================
2023-05-18 14:57:28.461Z: 

===================================== WARNING ===================================
2023-05-18 14:57:28.464Z: Creating recovery container.
=================================================================================

```

**Comment by techknowlogick** [comment]
@yardenshoham 
```
curl: (22) The requested URL returned error: 500 
```

nodejs.org is probably just having a tea break, hopefully it'll be back soon.

**Review by yardenshoham — APPROVED** [review]
It's a good base. Incomplete and we'll iterate. Next steps is auto `app.ini` creation and `make watch`.

**Review by wxiaoguang — APPROVED** [review]

**Pull request merged**